runtime/jcl/common/jclcinit.c

```diff
@@ -750,21 +749,20 @@ initializeRequiredClasses(J9VMThread *vmThread, char* dllName)
 	if (J9_ARE_ANY_BITS_SET(vm->extendedRuntimeFlags2, J9_EXTENDED_RUNTIME2_LOAD_AGENT_MODULE)
 		&& (NULL != vm->modulesPathEntry->extraInfo))
 	{
-		const char *module = NULL;
-		Trc_JCL_initializeRequiredClasses_addAgentModuleEntry(vmThread);
-		module = vm->jimageIntf->jimagePackageToModule(
+		const char *moduleName = vm->jimageIntf->jimagePackageToModule(
 				vm->jimageIntf, (UDATA) vm->modulesPathEntry->extraInfo,
 				"jdk/internal/agent");
-		if (NULL != module) {
+		if (NULL != moduleName) {
 			J9VMSystemProperty *systemProperty = NULL;
+			Trc_JCL_initializeRequiredClasses_addAgentModuleEntry(vmThread, moduleName);
 			/* Handle the case where there are no user-specified modules. In this case, there
 			 * is typically one system property but no user-set "add-modules" arguments.
 			 */
-			if ((0 == vm->addModulesCount) &&
-				(J9SYSPROP_ERROR_NONE == vmFuncs->getSystemProperty(vm, ADDMODS_PROPERTY_BASE "0", &systemProperty)))
+			if ((0 == vm->addModulesCount)
+				&& (J9SYSPROP_ERROR_NONE == vmFuncs->getSystemProperty(vm, ADDMODS_PROPERTY_BASE "0", &systemProperty)))
 			{
 				/* this is implicitly an unused property */
-				vmFuncs->setSystemProperty(vm, systemProperty, AGENT_MODULE_NAME);
+				vmFuncs->setSystemProperty(vm, systemProperty, moduleName);
```

Comment:
Not sure the VM should be loading the module if it's not the expected module "jdk.management.agent".

Reply:
What do you propose instead? An error?

Reply:
It doesn't need to be an error (can be a tracepoint), it just shouldn't load the module as the only requirements are to load the jdk.management.agent and/or ibm.healthcenter modules. If something else is found there is something funny going on.

Reply:
Ok, I will check that the module name is the expected one before requesting that it be loaded.
I'm not thrilled with the duplication of those module names, but I can live with it.

Reply:
Fixed in 5cd45ce0fea3.